src/main/java/oracle/kubernetes/operator/work/Fiber.java

```diff
@@ -231,6 +231,15 @@ public void resume(Packet resumePacket, CompletionCallback callback) {
       } finally {
         lock.unlock();
       }
+    } else if (status.get() == CANCELLED) {
```

Comment:
Let me propose a different fix here -- The problem with this fix is that the second fiber is hung until resume() is called on the first fiber, which might never happen.  Instead, let's move the exitCallback outside of completionCheck().

Reply:
The testing showed that the resume did happen on the first fiber for the use cases in the QA test cases. But I can try the solution that you suggested.